source/MaterialXCore/Util.cpp

```diff
@@ -179,4 +181,47 @@ string parentNamePath(const string& namePath)
     return EMPTY_STRING;
 }
 
+string normalizeNumericString(const string& str, unsigned int precision)
+{
+    std::stringstream ss(str);
+    std::stringstream result;
+    std::string token;
+
+    while (std::getline(ss, token, ','))
+    {
+        // Remove leading and trailing spaces
+        token.erase(0, token.find_first_not_of(' '));
+        token.erase(token.find_last_not_of(' ') + 1);
+
+        // Convert the string to a double (automatically removes leading zeros)
+        double number = std::stod(token);
```

Comment:
One important note is that we don't want to convert values strings to precision-limited numeric values in memory, and `normalizeNumericString` should be purely a string processing operation.  Given that framework, I wouldn't expect this function to take a `precision` argument, and the internal logic can be simplified to be purely string processing.

Reply:
If we don't do this conversion to a numeric type, then the string processing should probably also removing any leadings zeros as well, as well as extraneous `+` signs.  Converting to the concrete numeric type probably actually makes this code a little simpler, but at the expense of the precision conversion @jstone-lucasfilm points out.... 

Even if this function takes a purely string manipulation approach to the normalization, I can still imagine times where it might be useful to be able to normalize the precision of the numeric values in the document. I've seen documents where some automatic process has written out a value of say `53.9999999999` instead of `54`, and with a `precision` argument to the function, the document could be made more human readable again, but again if we're going to take the purely string manipulation approach here that will likely make the code yet more complex.

Reply:
Thanks for pointing this out. I added precision only because I was using double conversion.
Logic should be simpler now.

Reply:
There are ways to control what precision to access a number with and they could be used to overwrite existing values. I think it's better to keep that separate from this utility which should not change the value stored.